Files/Helpers/ExternalResourcesHelper.cs

```diff
@@ -117,12 +118,39 @@ public async Task<bool> TryLoadThemeAsync(AppTheme theme)
                 return false;
             }
         }
+
+        public async void UpdateTheme(AppTheme OldTheme, AppTheme NewTheme)
+        {
+            if (OldTheme.Path != null)
+            {
+                RemoveTheme(OldTheme);
+            }
+
+            if(NewTheme.Path != null)
+            {
+                await TryLoadThemeAsync(NewTheme);
+            }
+        }
+
+        public bool RemoveTheme(AppTheme theme)
+        {
+            try
+            {
+                App.Current.Resources.MergedDictionaries.Remove(App.Current.Resources.MergedDictionaries.Where(x => x.TryGetValue("CustomThemeID", out var key) && (key as string) == theme.Key).FirstOrDefault());
```

Comment:
```suggestion
                App.Current.Resources.MergedDictionaries.Remove(App.Current.Resources.MergedDictionaries.FirstOrDefault(x => x.TryGetValue("CustomThemeID", out var key) && (key as string) == theme.Key));
```
Does something like this work in place of two LINQ calls?

Reply:
@winston-de

Reply:
Yes it would, I didn't know that was possible thanks for pointing it out.